intelmq/bots/outputs/templated_smtp/output.py

```diff
@@ -0,0 +1,187 @@
+from intelmq.lib.exceptions import ConfigurationError
+from intelmq.lib.exceptions import MissingDependencyError
+
+
+class TemplatedSMTPOutputBot(Bot):
+    smtp_host: Optional[str] = None
+    smtp_port: int = 25
+    ssl: bool = False
+    starttls: bool = False
+    username: Optional[str] = None
+    password: Optional[str] = None
+    verify_cert: bool = True
+    attachments: List[str] = []
+    mail_from: Optional[str] = None
+    mail_to: Optional[str] = None
+    subject: Optional[str] = None
+    body: Optional[str] = None
+
+    def init(self):
+        if not Template:
+            raise MissingDependencyError("jinja2")
+
+        if not self.smtp_host:
+            raise ConfigurationError("Server", "No SMTP host specified")
+        if self.ssl:
+            self.smtp_class = smtplib.SMTP_SSL
+        else:
+            self.smtp_class = smtplib.SMTP
+        if not self.mail_from:
+            raise ConfigurationError("Mail", "No sender specified")
+        if not self.mail_to:
+            raise ConfigurationError("Mail", "No recipient specified")
+        if not self.subject:
+            raise ConfigurationError("Mail", "No subject specified")
+        if not self.body:
+            raise ConfigurationError("Mail", "No body specified")
+
+        self.templates = {
+            "subject": Template(self.subject),
+            "from": Template(self.mail_from),
+            "to": Template(self.mail_to),
+            "body": Template(self.body),
+            "attachments": []
+        }
+        for att in self.attachments:
+            if "content-type" not in att:
+                self.logger.error("Attachment does not have a content-type, ignoring: %s.", att)
+                continue
+            elif "text" not in att:
+                self.logger.error("Attachment does not have a text, ignoring: %s.", att)
+                continue
+            elif "name" not in att:
+                self.logger.error("Attachment does not have a name, ignoring: %s.", att)
+                continue
+            attachment_template = {
+                "content-type": Template(att["content-type"]),
+                "text": Template(att["text"]),
+                "name": Template(att["name"])
+            }
+            self.templates["attachments"].append(attachment_template)
+
+    def process(self):
+        event = self.receive_message()
+
+        if self.verify_cert and self.smtp_class == smtplib.SMTP_SSL:
+            kwargs = {'context': ssl.create_default_context()}
+        else:
+            kwargs = {}
+
+        with self.smtp_class(self.smtp_host, self.smtp_port,
+                             **kwargs) as smtp:
+            if self.starttls:
+                if self.verify_cert:
+                    smtp.starttls(context=ssl.create_default_context())
+                else:
+                    smtp.starttls()
+            if self.username and self.password:
+                smtp.login(user=self.username, password=self.password)
+
```

Comment:
The `smtp.login` call could use some exception handling and debug logging.
And maybe we could check in the `init` if both username and password are set

Reply:
Same as above regarding exception handling.